test/functional/bulk_tests.js

```diff
@@ -937,68 +937,59 @@ describe('Bulk', function() {
+      poolSize: 1
+    });
 
-          // Add unique index on b field causing all updates to fail
-          col.ensureIndex({ b: 1 }, writeConcern, function(err) {
-            expect(err).to.not.exist;
+    client.connect(function(err, client) {
+      const db = client.db(configuration.db);
+      const col = db.collection('err_batch_write_unordered_ops_legacy_6');
+
+      // Add unique index on a field causing all inserts to fail
+      col.createIndexes(
+        [
+          {
+            name: 'err_batch_write_unordered_ops_legacy_6',
+            key: { a: 1 },
+            unique: true
+          }
+        ],
+        function(err) {
+          expect(err).to.not.exist;
 
-            // Initialize the unordered Batch
-            var batch = col.initializeUnorderedBulkOp();
+          // Initialize the unordered Batch
+          const batch = col.initializeUnorderedBulkOp();
 
-            // Add some operations to be executed in order
-            batch.insert({ a: 1 });
-            batch.find({ a: 1 }).update({ $set: { b: 1 } });
-            batch.insert({ b: 1 });
-            batch.insert({ b: 1 });
-            batch.insert({ b: 1 });
-            batch.insert({ b: 1 });
+          // Add some operations to be executed in order
+          batch.insert({ a: 1 });
+          batch.insert({ a: 1 });
 
-            // Execute the operations
-            batch.execute(configuration.writeConcernMax(), function(err, result) {
-              expect(err).to.exist;
-              expect(result).to.not.exist;
+          // Execute the operations
+          batch.execute(configuration.writeConcernMax(), function(err, result) {
+            expect(err).to.exist;
+            expect(result).to.not.exist;
 
-              // Test basic settings
-              result = err.result;
-              expect(result.nInserted).to.equal(2);
-              expect(result.hasWriteErrors()).to.equal(true);
-              expect(
-                result.getWriteErrorCount() === 4 || result.getWriteErrorCount() === 3
-              ).to.equal(true);
+            // Test basic settings
+            result = err.result;
+            expect(result.nInserted).to.equal(1);
+            expect(result.hasWriteErrors()).to.equal(true);
+            expect(result.getWriteErrorCount() === 1).to.equal(true);
 
-              // Individual error checking
-              var error = result.getWriteErrorAt(0);
-              expect(error.code === 11000 || error.code === 11001).to.equal(true);
-              expect(error.errmsg).to.exist;
-              expect(err.message).to.equal(error.errmsg);
+            // Individual error checking
+            const error = result.getWriteErrorAt(0);
+            expect(error.code === 11000).to.equal(true);
+            expect(error.errmsg).to.exist;
+            expect(err.message).to.equal(error.errmsg);
 
-              client.close();
-              done();
-            });
+            client.close();
```

Comment:
Let's change this to `client.close(done);` since we're here. This helps us modernize our tests.